### PR TITLE
Alternative method to Utils optional options

### PR DESCRIPTION
**Pull request by GeoDerp**

### CODE REVIEW

**Comment by GeoDerp** [comment]
@davidusb-geek 
You can pick one or the other. 
Personally I like this approach more as its easy to read. However mat be complex to understand.

**Comment by davidusb-geek** [comment]
This option seems much more elegant. Nice work.

**Comment by davidusb-geek** [comment]
> Both solar.forcast and solcast should be broken. That was the main reason for both pull requests initially. Would have to check if that's related to that comment however.

The bugs from this comment are also solved on this PR.
In other words, if we use this PR instead of #167, the bugs on solar.forcast and solcast are solved?

**Comment by GeoDerp** [comment]
In testing it seemed to fix the issue also. Will do a double check now to be double sure.

**Comment by davidusb-geek** [comment]
I think that we will push up this method, if everything works correctly.

**Comment by GeoDerp** [comment]
Tested, looks like everything works. Actually found an bug when setting solar_forecast_kwp to 0 so added a patch fix (feell free to revert)

**Comment by GeoDerp** [comment]
hangon. may of found another bug. Looking at it now

**Comment by GeoDerp** [comment]
Should be good to go now 👍 
My most tested code yet

**Comment by davidusb-geek** [comment]
Great! +1

**Review by davidusb-geek — APPROVED** [review]

File: src/emhass/utils.py [hunk 1]
```diff
@@ -573,114 +573,97 @@ def build_params(params: dict, params_secrets: dict, options: dict, addon: int,
...
-        params['plant_conf']['Pd_max'] = options.get('battery_discharge_power_max',params['plant_conf']['Pd_max']) 
-        params['plant_conf']['Pc_max'] = options.get('battery_charge_power_max',params['plant_conf']['Pc_max'])
-        params['plant_conf']['eta_disch'] = options.get('battery_discharge_efficiency',params['plant_conf']['eta_disch'])
-        params['plant_conf']['eta_ch'] = options.get('battery_charge_efficiency',params['plant_conf']['eta_ch'])
-        params['plant_conf']['Enom'] = options.get('battery_nominal_energy_capacity',params['plant_conf']['Enom'])
-        params['plant_conf']['SOCmin'] = options.get('battery_minimum_state_of_charge',params['plant_conf']['SOCmin']) 
-        params['plant_conf']['SOCmax'] = options.get('battery_maximum_state_of_charge',params['plant_conf']['SOCmax']) 
-        params['plant_conf']['SOCtarget'] = options.get('battery_target_state_of_charge',params['plant_conf']['SOCtarget'])
+
+        params['associations_dict'] = associations_dict
```
[the remark this answers is not in the document]
Comment (GeoDerp) [reply src/emhass/utils.py:668]: We do. For testing. We check that options parameters are properly overriding the default config

**Pull request merged**